python/cucim/src/cucim/skimage/segmentation/_chan_vese.py

```diff
@@ -429,18 +437,14 @@ def chan_vese(
     if extended_output:
         old_energy = _cv_energy(image, phi, mu, lambda1, lambda2)
         energies = []
-    phivar = tol + 1
 
-    dt = cp.asarray(dt, dtype=float_dtype)
-    mu = cp.asarray(mu, dtype=float_dtype)
-    lambda1 = cp.asarray(lambda1, dtype=float_dtype)
-    lambda2 = cp.asarray(lambda2, dtype=float_dtype)
+    phivar = tol + 1
     while phivar > tol and i < max_num_iter:
```

Comment:
As we compare `phivar` to `tol`, it helps if they are both in the same location (on device or on host). Given `phivar` will be on device (as noted above), it makes sense to have `tol` on device too